### ISSUE TITLE
`SingleCellExperiment`/`Seurat` should contain `dgCMatrix` instead of `dgRMatrix`

### ISSUE THREAD

**Issue by pakiessling** [scverse/anndataR#202]
Similiar to the discussion in https://github.com/theislab/zellkonverter/issues/34 

A lot of tools like https://github.com/immunogenomics/presto/ will try to call functions from the Matrix package that error when encountering a dgRMatrix.

**Comment by Artur-man** [comment]
Looking at my code, I hardly use dgRMatrix, most of my converted matrices are in dgCMatrix

**Comment by pakiessling** [comment]
I use anndataR to integrate R methods in my mostly python based workflow.

I will routinely read data from on-disk .h5ads into Seurat / SingleCellExperiment to then apply tools e.g. [presto](https://github.com/immunogenomics/presto/ )

These tools (Is suspect all tools calling common {Matrix} functions) will then crash with an error message about the matrix having the wrong format.

I would expect a Seurat / SingleCellExperiment object generated by anndataR to be out-of-the-box compatible with functions taking Seurat / SingleCellExperiment objects as input

**Comment by pakiessling** [comment]
Yes here is what happened:

```
library(anndataR)
library(presto)

sce = read_h5ad("adata.h5ad",  to="SingleCellExperiment)
wilcoxauc(sce, 'cell_type')
> Error message about incorrect matrix format.
```
I think that this should "just work".
Less sure about how to handle things when staying inside an anndata, but isnt the main draw of using anndataR to be able to use R packages on anndatas? Won't a lot of them always fail for `dgRMatrix` input?

**Comment by lazappi** [comment]
Thanks for the extra detail. I will update the issue title to make it a bit clearer. @rcannood @LouiseDck this is pretty important for anyone who wants to read a `.h5ad` to a `SingleCellExperiment`/`Seurat`. As part of the conversion when matrices are transposed they also need to be converted from `dgRMatrix` to `dgCMatrix`.

**Comment by LouiseDck** [comment]
I was following the discussion here, and read the issue on `{zellkonverter}`. I am convinced that the conversion should probably produce `dgCMatrices`. 

I am not sure whether they should be `dgCMatrices` in the anndata representation --> if you want to interact with the matrices in the anndata object, you might want to run more specific methods where a row major format makes more sense? On the other hand, if you want to run these specific methods, you might be more aware of the differences and be able to easily convert it by yourself?

Do you have any thoughts @rcannood ?

**Comment by lazappi** [comment]
I guess `dgRMatrix` matches how the data is arranged in `AnnData`. Currently no analysis packages support `AnnData` so getting the conversion right is the important thing and I'm still pretty strongly of the belief we should discourage most users from interacting with R `AnnData` directly to avoid creating another R data format.

**Comment by rcannood** [comment]
I agree SCE and Seurat should receive a dgCMatrix if they're not able to process matrices in other formats. Let's create a PR for this!

The InMemoryAnnData or HDF5AnnData should remain unchanged -- if the obsm is a dgRMatrix and would need to be converted to dgCMatrix for some functions to work, it's up to the user to take care of that.

------

>  I'm still pretty strongly of the belief we should discourage most users from interacting with R AnnData directly to avoid creating another R data format.

I beg to differ -- if anndataR can do all the things dynverse/anndata on CRAN can, I'd be happy to deprecate dynverse/anndata in favour for anndataR so there is one package less to maintain ;) But this is a separate discussion.

**Comment by lazappi** [comment]
reprex confirming this is still the case on `main`:

``` r
library(anndataR)

adata <- generate_dataset(
  format = "AnnData",
  x_type = "numeric_rsparse",
  layer_types = c(),
  obs_types = c("character"),
  var_types = c("character"),
  obsm_types = c(),
  varm_types = c(),
  obsp_types = c(),
  varp_types = c(),
  uns_types = c()
)
write_h5ad(adata, "test_adata.h5ad")

sce <- read_h5ad("test_adata.h5ad", to = "SingleCellExperiment")
class(SummarizedExperiment::assay(sce, "counts"))
#> [1] "dgRMatrix"
#> attr(,"package")
#> [1] "Matrix"

seurat <- read_h5ad("test_adata.h5ad", to = "Seurat")
#> Warning: Data is of class dgRMatrix. Coercing to dgCMatrix.
class(SeuratObject::LayerData(seurat, "counts"))
#> [1] "dgCMatrix"
#> attr(,"package")
#> [1] "Matrix"
```

<sup>Created on 2025-03-06 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**{Seurat}** gives a warning about doing the conversion but it would probably be better to do it on our side to avoid that

**Comment by LouiseDck** [comment]
This should be fixed in #258
